tests/unitary/src/compute/allocator/test_cuda_memory_block_pool.cpp

```diff
@@ -0,0 +1,521 @@
+    REQUIRE( ite->first.get_queue_id() == 16 );
+    REQUIRE( ite->second.get_next_block() == null );
+    REQUIRE( ite->second.is_free() == false );
+    REQUIRE( check_backward_link(ite) == true );
+}
+
+TEST_CASE( "consider_merging_block should not merge when is not partition", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 1);
+
+    consider_merging_block(pool, ite);
+    REQUIRE( pool.size() == 1 );
+}
+
+TEST_CASE( "consider_merging_block should not merge when prev is occupied", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 2);
+    ite->second.set_free(false);
+    ite = ite->second.get_next_block();
+
+    const auto old_ite = ite;
+    consider_merging_block(pool, ite);
+    REQUIRE( ite == old_ite );
+    REQUIRE( pool.size() == 2 );
+}
+
+TEST_CASE( "consider_merging_block should not merge when next is occupied", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 2);
+    ite->second.get_next_block()->second.set_free(false);
+
+    const auto old_ite = ite;
+    consider_merging_block(pool, ite);
+    REQUIRE( ite == old_ite );
+    REQUIRE( pool.size() == 2 );
+}
+
+TEST_CASE( "consider_merging_block should merge when prev is free", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 2);
+    ite->second.set_free(true);
+    ite = ite->second.get_next_block();
+    ite->second.set_free(true);
+
+    consider_merging_block(pool, ite);
+    REQUIRE( pool.size() == 1 );
+}
+
+TEST_CASE( "consider_merging_block should merge when next is free", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 2);
+    ite->second.set_free(true);
+    ite->second.get_next_block()->second.set_free(true);
+
+    consider_merging_block(pool, ite);
+    REQUIRE( pool.size() == 1 );
+}
+
+TEST_CASE( "consider_merging_block should merge when prev and next is free", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 3);
+    ite->second.set_free(true);
+    ite = ite->second.get_next_block();
+    ite->second.set_free(true);
+    ite->second.get_next_block()->second.set_free(true);
+
+    consider_merging_block(pool, ite);
+    REQUIRE( pool.size() == 1 );
+}
+
+TEST_CASE( "merge_blocks (2) produce valid blocks", "[cuda_memory_block_pool]" )
+{
+    cuda_memory_block_pool pool;
+    auto ite = generate_partitions(pool, 4);
+
+    const auto keep_left = ite;
+    const auto merge1 = keep_left->second.get_next_block();
+    const auto merge2 = merge1->second.get_next_block();
+    const auto keep_right = merge2->second.get_next_block();
+
+    const auto merged = merge_blocks(pool, merge1, merge2);
+    REQUIRE( reinterpret_cast<std::uintptr_t>(merged->first.get_data()) == (0xDEADBEEF + 1024));
```

Comment:
where is 0xDEADBEEF coming from

Reply:
Replaced to using a constant instead of hardcoded values